src/parser/metrics/Complexity.ts

```diff
@@ -44,12 +44,94 @@ export class Complexity implements Metric {
                     }
                 }
             }
-        });
+        }
+        this.createCaseQueryStatements(allNodeTypes);
+    }
+
+    createCaseQueryStatements(allNodeTypes: ExpressionMetricMapping[]) {
```

Comment:
I recommend to refactor this function to make it more readable. It's complexity is roughly 12 which is a little bit too high.

Reply:
I have already splitted out a part of this now, as the length of this function further escalated as I added a special treatment for Kotlins `when_entry`

Reply:
Ah, ok, you meant the constructor. I will check that.

Reply:
Refactored the constructor and saved one loop over all node types, which should also improve the performance.